### DISCUSSION TITLE
FastAPI - get return variable from different file

### DISCUSSION THREAD

**Discussion by Kammmil** [fastapi/fastapi#8563] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
"""file1"""

from fastapi import FastAPI

app = Fast API(title='myapp', docs_url='/', description="some_desc")
# have tried also adding below line
# app.store = None

def run_fastapi(context):
    uvicorn.run("api:app", reload=False)

@app.get("/api/some_value", responses=SOME_RESPONSES)
def get_parcels(response: Response):
    test = app.store
    print(test)


"""file2"""

import file1

if __name__  == "__main__":
  file1.app.store = "Some Value"
  file1.app.run_fastapi()
```

### Description

I got a file with Fast API,
pretty simple thing that all it does is have 1 get that should return a variable.
Then I am trying to import that app in different file and pass variable that is calculated in that file to Fast API app.
So pretty much I need it to work like that:
Fast_API is getting "get" request, it checks what value is currently of app.store variable in file1 and then return it over that get request.
Got no idea how to make it work in Fast API since it's totally different in Flask.
Thank you for your time, and I will be happy to help others.

### Operating System

Linux

### FastAPI Version

0.70.0

### Python Version

3.6.9

**Comment by jgould22** [comment]
Have you tried something like 
```python
from fastapi import FastAPI, Request

app = Fast API(title='myapp', docs_url='/', description="some_desc")
# have tried also adding below line
# app.store = None

def run_fastapi(context):
    uvicorn.run("api:app", reload=False)

@app.get("/api/some_value", responses=SOME_RESPONSES)
# Why would your controller here take response as a parameter?
# You probably want to use Request instead which should give you access to the app
def get_parcels(request: Request):
    test = request.app.store
    print(test)

```

**Comment by Kammmil** [comment]
```
from fastapi import FastAPI, Request

app = Fast API(title='myapp', docs_url='/', description="some_desc")
app.test = 'in file1'

def run_fastapi(context):
    uvicorn.run("api:app", reload=False)

@app.get("/api/some_value", responses=SOME_RESPONSES)
def get_parcels(request: Request):
    test = request.app.store
    print(test)
```
and then in file 2

```
import file1

if __name__  == "__main__":
  file1.app.store = "In file 2"
  file1.app.run_fastapi()
```
is printing "in file1" and I need it to print "in file2"

**Comment by jgould22** [comment, marked as answer]
What about if you did something like 

```python 
# file1.py
from fastapi import FastAPI, Request
import uvicorn

app = FastAPI(title='myapp', docs_url='/', description="some_desc")
app.state = "In file 1"

def update_state(string):
    app.state =  string

def run_fastapi():
    uvicorn.run(app, reload=False)

@app.get("/api/some_value")
def get_parcels(request: Request):
    test = request.app.state
    print(test)
```
and
```python
# file2.py
import file1

if __name__  == "__main__":
  print(file1.app.state)
  file1.update_state("In file 2")
  file1.run_fastapi()

**Comment by Kammmil** [comment]
It does work right now, thank you :)
